### PR TITLE
Alternative logging

### PR DESCRIPTION
**Pull request by n8wb**

### CODE REVIEW

File: protocols/registrar/registrar.go [hunk 1]
```diff
@@ -3,17 +3,17 @@
...
-    GNU General Public License for more details.
+	Genesis is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
 
-    You should have received a copy of the GNU General Public License
-    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+	You should have received a copy of the GNU General Public License
+	along with this program.  If not, see <https://www.gnu.org/licenses/>.
```
Comment (atoulme) [inline protocols/registrar/registrar.go]: can we cancel out those changes?

File: testnet/testnet.go [hunk 2]
```diff
@@ -219,7 +223,7 @@ func (tn *TestNet) AddDetails(dd db.DeploymentDetails) error {
 		}
 	}
 	for i := len(tn.CombinedDetails.Resources); i < oldCD.Nodes; i++ {
-		tn.CombinedDetails.Resources = append(tn.CombinedDetails.Resources, tn.GetNodeResources(i))
+		tn.CombinedDetails.Resources = append(tn.CombinedDetails.Resources, tn.GetNodeResources(tn.Nodes[i]))
```
Comment (atoulme) [inline testnet/testnet.go]: Is there a reason for that change?

File: docker/container.go [hunk 3]
```diff
@@ -3,17 +3,17 @@
...
-    GNU General Public License for more details.
+	Genesis is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
 
-    You should have received a copy of the GNU General Public License
-    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+	You should have received a copy of the GNU General Public License
+	along with this program.  If not, see <https://www.gnu.org/licenses/>.
```
Comment (atoulme) [inline docker/container.go]: same here

File: protocols/registrar/registrar.go (hunk 1 above)
Comment (n8wb) [reply protocols/registrar/registrar.go]: We are inconsistent with our GPL headers, sometimes we use tabs, sometimes spaces, so I just converted this one to tabs like the rest of them

File: docker/container.go (hunk 3 above)
Comment (n8wb) [reply docker/container.go]: We are inconsistent with our GPL headers, sometimes we use tabs, sometimes spaces, so I just converted this one to tabs like the rest of them

File: testnet/testnet.go (hunk 2 above)
Comment (n8wb) [reply testnet/testnet.go]: Yeah, i changed the interface so that it can extract information from arbitrary nodes, rather than having to add them first

**Pull request merged**